### ISSUE TITLE
write shared ignores format keyword

### ISSUE THREAD

**Issue by barronh** [Unidata/netcdf4-python#827]
Adding `s` to modes `a`, `r+`, or `w` is expected to only affect `NETCDF3` files. If I specify `ws` for a `NETCDF4_CLASSIC` file, I expect the `s` to be ignored. Instead, whichever format was created last is created again.

```
from netCDF4 import Dataset
f3 = Dataset('test3.nc', format='NETCDF3_CLASSIC', mode='ws')
print(f3.data_model)
f3l = Dataset('test3l.nc', format='NETCDF3_64BIT_OFFSET', mode='ws')
print(f3l.data_model)
f4 = Dataset('test4.nc', format='NETCDF4_CLASSIC', mode='ws')
print(f4.data_model)
```

Results
```
NETCDF3_CLASSIC
NETCDF3_CLASSIC
NETCDF3_CLASSIC
```

Expected

```
NETCDF3_CLASSIC
NETCDF3_64BIT_OFFSET
NETCDF4_CLASSIC
```

Looks like `ws` is missing the `_set_default_format(format=format)` command.

Thanks

**Comment by jswhit** [comment]
Right - that's a bug.  Fixed now in pull request #825.  

Note however that the 's' flag will be ignored for HDF5 formatted files (like NETCDF4_CLASSIC).

**Comment by jswhit** [comment]
pull request #825 merged, closing now
